File: src/main/java/scrabble/phrases/words/Word.java
package scrabble.phrases.words;

/**
 * Sealed interface for all word types in the dictionary.
 */
public sealed interface Word permits Noun, Adjective, Verb {

    /**
     * Gets the word text.
     *
     * @return the word
     */
    String word();

    /**
     * Gets the number of syllables.
     *
     * @return syllable count
     */
    int syllables();

    /**
     * Gets the rhyme suffix (last 3 characters).
     *
     * @return the rhyme
     */
    String rhyme();
}

File: src/main/kotlin/scrabble/phrases/words/Word.kt
package scrabble.phrases.words

sealed interface Word {
    val word: String
    val syllables: Int
    val rhyme: String
}

/** Extension property giving read access to [Word.syllables] under a stable name.
 *  Returns the same value — never returns null, never deviates from `syllables`.
 */
val Word.syllableCount: Int
    get() = this.syllables
